client/engine/chainservice/eth_chainservice.go

```diff
@@ -123,8 +124,17 @@ func (ecs *EthChainService) listenForLogEvents(na *NitroAdjudicator.NitroAdjudic
 					panic(err)
 				}
 
-				// TODO set Holdings based on chain event
-				event := AllocationUpdatedEvent{CommonEvent: CommonEvent{channelID: au.ChannelId, BlockNum: chainEvent.BlockNumber}, Holdings: types.Funds{}}
+				tx, pending, err := chain.TransactionByHash(context.Background(), chainEvent.TxHash)
+				if pending || err != nil {
+					panic("Expected transacion to be part of the chain")
+				}
+
+				assetAddress, amount, err := getChainHolding(na, tx, au)
+				if err != nil {
+					panic(err)
+				}
```

Comment:
Do we want to panic here? What kind of errors might we expect?

Reply:
Do you have suggestion on what the code should do instead? An error here means that we attempted to read the amount of tokens held for a channel in the adjudicator and failed to do so.

At the moment, the chain service is panic-happy. Down the line, a better approach might be for the chain service to emit errors via a channel and allow other components to handle the errors.